server/src/main/java/org/candlepin/controller/ContentManager.java

```diff
@@ -525,51 +527,147 @@ public Content updateContent(ContentData update, Owner owner, boolean regenerate
+        if (content != null && !content.isEmpty()) {
+            Map<String, Content> contentMap = new HashMap<String, Content>();
+            for (Content entity : content) {
+                contentMap.put(entity.getUuid(), entity);
             }
+
+            this.removeContentByUuids(owner, contentMap.keySet(), regenerateEntitlementCerts);
         }
+    }
+
+    /**
+     * Removes all content from the given owner.
+     *
+     * @param owner
+     *  The owner from which to remove content
+     *
+     * @param regenerateEntitlementCerts
+     *  Whether or not changes made to the content should trigger the regeneration of entitlement
+     *  certificates for affected consumers
+     *
+     * @throws IllegalArgumentException
+     *  if owner is null
+     */
+    public void removeAllContent(Owner owner, boolean regenerateEntitlementCerts) {
+        this.removeContentByUuids(owner, this.ownerContentCurator.getContentUuidsByOwner(owner),
+            regenerateEntitlementCerts);
+    }
+
+    /**
+     * Removes all content with the provided UUIDs from the given owner.
+     *
+     * @param owner
+     *  The owner from which to remove content
+     *
+     * @param contentUuids
+     *  A collection of UUIDs representing the content to remove
+     *
+     * @param regenerateEntitlementCerts
+     *  Whether or not changes made to the content should trigger the regeneration of entitlement
+     *  certificates for affected consumers
+     *
+     * @throws IllegalArgumentException
+     *  if owner is null
+     */
+    public void removeContentByUuids(Owner owner, Collection<String> contentUuids,
+        boolean regenerateEntitlementCerts) {
 
-        this.ownerContentCurator.removeOwnerFromContent(existing, owner);
-        if (this.ownerContentCurator.getOwnerCount(existing) == 0) {
-            // No one is using this product anymore; delete the entity (should clean up everything)
-            this.contentCurator.delete(existing);
+        if (owner == null) {
+            throw new IllegalArgumentException("owner is null");
         }
-        else {
-            // Clean up any dangling references to content
-            this.ownerContentCurator.removeOwnerContentReferences(existing, owner);
+
+        if (contentUuids != null && !contentUuids.isEmpty()) {
+            log.debug("Deleting content with UUIDs: {}", contentUuids);
+
+            List<Product> affectedProducts = this.productCurator
+                .getProductsByContentUuids(owner, contentUuids)
+                .list();
+
+            if (!affectedProducts.isEmpty()) {
+                log.debug("Updating {} affected products", affectedProducts.size());
+
+                if (!(contentUuids instanceof Set)) {
+                    // Convert this to a set so our filtering lookups aren't painfully slow
+                    contentUuids = new HashSet<String>(contentUuids);
+                }
```

Comment:
A lot of our other stuff/queries return(s( lists of entities, so as much as I'd like forcing a specific type of collection, it'd just make the method itself more painful to use.